io.openems.edge.evcs.goe.chargerhome/src/io/openems/edge/evcs/goe/chargerhome/GoeChargerHomeImpl.java

```diff
@@ -0,0 +1,340 @@
+				this.channel(GoeChannelId.CURRENT_L2).setNextValue(nrg.get(5).getAsInt()*100);
+				this.channel(GoeChannelId.CURRENT_L3).setNextValue(nrg.get(6).getAsInt()*100);
+				this.channel(GoeChannelId.ACTUAL_POWER).setNextValue(nrg.get(11).getAsInt()*10);
+		
+				this.channel(Evcs.ChannelId.PHASES).setNextValue(phases);				
+				this.channel(Evcs.ChannelId.MAXIMUM_POWER).setNextValue(GetMaxPower(phases));	
+				this.channel(Evcs.ChannelId.MINIMUM_HARDWARE_POWER).setNextValue(GetMinHardwarePower(phases, cabelCurrent));	
+				this.channel(Evcs.ChannelId.MAXIMUM_HARDWARE_POWER).setNextValue(GetMaxHardwarePower(cabelCurrent));	
+				this.channel(Evcs.ChannelId.CHARGE_POWER).setNextValue(nrg.get(11).getAsInt()*10);
+				this.channel(Evcs.ChannelId.CHARGING_TYPE).setNextValue(ChargingType.AC);
+				
+				this.channel(GoeChannelId.ENERGY_TOTAL).setNextValue(json.get("eto").getAsInt()*100);				
+				this.channel(Evcs.ChannelId.ENERGY_SESSION).setNextValue(json.get("dws").getAsInt()*10/3600);
+			
+				this.channel(GoeChannelId.ERROR).setNextValue(err);
+				this.channel(Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED).setNextValue(false);
+				
+				this.setEnergySession();
+				this.setPower();				
+				
+			}
+			else{
+				this.channel(Evcs.ChannelId.CHARGINGSTATION_COMMUNICATION_FAILED).setNextValue(true);
+			}
+			
+			break;
+		}
+	}
+	
+	private int ConvertGoeStatus(int status) {
+		switch(status) {
+		case 1: // ready for charging, car unplugged
+			return Status.NOT_READY_FOR_CHARGING.getValue();
+		case 2: // charging
+			return Status.CHARGING.getValue();
+		case 3: // waiting for car
+			return Status.READY_FOR_CHARGING.getValue(); 
+		case 4: // charging finished, car plugged
+			return Status.CHARGING_FINISHED.getValue();
+		default:
+			return Status.UNDEFINED.getValue();
+		}
+	}
+	
+	private int ConvertGoePhase(int phase) {
+		int Phasen = (byte)phase & 0b00111000;			
+		switch(Phasen) {
+		case 8: // 0b00001000: Phase 1 is active
+			return 1;
+		case 24: // 0b00011000: Phase 1+2 is active
+			return 2;
+		case 56: // 0b00111000: Phase1-3 are active
+			return 3;
+		default:
+			return 0;
+		}
+	}
+	
+	private int GetMaxHardwarePower(int cableCurrent) {
+		int MaxPower = 0;
+		int phases = 3;
+		if (cableCurrent < this.MaxCurrent && cableCurrent > 0) {
+			MaxPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MaxPower = phases*this.MaxCurrent*230/1000;
+		}
+		return MaxPower;
+	}
+	
+	private int GetMinHardwarePower(int phases, int cableCurrent) {
+		int MinPower = 0;
+		int MinPowerHw = phases*this.MinCurrent*230/1000;
+		if (cableCurrent < this.MinCurrent && cableCurrent > 0) {
+			MinPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MinPower = phases*this.MinCurrent*230/1000;
+		}
+		if (MinPower > MinPowerHw || MinPower < 0) {
+			return MinPowerHw;
+		}
+		else {
+			return MinPower;
+		}
+	}
+		
+	private int GetMaxPower(int phases) {
+		return phases*this.MaxCurrent*230/1000;
```

Comment:
Should also be limited by the Go-E Charger Hardware:
11kW Go-e Home Charger: 16A --> 230V * 16A * 3 = 11.040 W 
22kW Go-e Home Charger: 32A --> 230V * 32A * 3 = 22.080 W

So for the 11kW Charger the calculated max Power would be 22.080W (22kW) because the default "maxCurrent" is 32000mA also the real maxCurrent for a 11kW Charger is 16A

Reply:
This method is anyways not needed, but if we are not getting the real maximum current 11 or 22 kW charger from the charger itself, we have only the possibility to get this value out of the configuration

Reply:
Im not sure on that but i think somewhere deep deep in the API is a value (i will lookup that!)

But if not is it possible to make a Select-Field so that the User itself (the user must know witch charger he got) can Select his Type when he adds the Go-E charger?

Reply:
We are pre configuring such things for customers, but of course it would be great to have the direct infromation from the charger

Reply:
I just looked up the Api Doc at GH and found that

![Screenshot_20210623_184030_com android chrome](https://user-images.githubusercontent.com/60319091/123135906-a2e11980-d452-11eb-9330-2849a40806f2.jpg)

If my understanding is right the cbl parameter gives Informations about how much Current is the max Power

Reply:
yes, that's right. My cable allows 32A and status for the parameter is "cbl": "32".
If you use a 22kW Charger with a 16A Cable the maximum Hardware-Power is 11kW. So i took the maximum current from configuration and limit the value with the used cable.